xblockutils/show_answers.py

```diff
@@ -0,0 +1,118 @@
+"""XBlock mixins for answers"""
+import logging
+
+from django.utils.translation import ugettext_lazy as _
+from xblock.fields import (
+    Scope,
+    String,
+)
+
+from xblockutils.constants import ShowAnswer
+
+
+log = logging.getLogger(__name__)
+
+
+class ShowAnswerXBlockMixin(object):
+    """
+    Mixin for showanswer functionality
+    """
+
+    showanswer = String(
+        display_name=_("Show Answer"),
+        help=_("Defines when to show the answer to the problem. "
+               "A default value can be set in Advanced Settings."),
+        scope=Scope.settings,
+        default=ShowAnswer.PAST_DUE,  # Default to PAST_DUE instead of FINISHED for backwards compat
+        values=[
+            {"display_name": _("Always"), "value": ShowAnswer.ALWAYS},
+            {"display_name": _("Answered"), "value": ShowAnswer.ANSWERED},
+            {"display_name": _("Attempted"), "value": ShowAnswer.ATTEMPTED},
+            {"display_name": _("Closed"), "value": ShowAnswer.CLOSED},
+            {"display_name": _("Finished"), "value": ShowAnswer.FINISHED},
+            {"display_name": _("Correct or Past Due"), "value": ShowAnswer.CORRECT_OR_PAST_DUE},
+            {"display_name": _("Past Due"), "value": ShowAnswer.PAST_DUE},
+            {"display_name": _("Never"), "value": ShowAnswer.NEVER}
+        ]
+    )
+    solution = String(
+        help=_("Solution to the problem to show to the user"),
+        scope=Scope.settings,
+        multiline_editor='html',
+        resettable_editor=False,
+    )
+
+    def answer_available(self):  # pylint: disable=too-many-return-statements
```

Comment:
This is adapted from https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/capa_base.py#L892